### PR TITLE
Add new tokens

### PR DESCRIPTION
**Pull request by lthiery**
New tokens are implemented.

A "Generic" type is created to assist in parsing inputs when we don't know which token it will be.

~~Note: tests will not pass until mainnet API provides compatible JSON.~~ _mainnet API has been updated_

### CODE REVIEW

File: src/models/values.rs [hunk 1]
```diff
@@ -115,6 +115,9 @@ macro_rules! decimal_scalar {
 
 decimal_scalar!(Hnt, 100_000_000, 8);
 decimal_scalar!(Hst, 100_000_000, 8);
+decimal_scalar!(Iot, 100_000_000, 8);
+decimal_scalar!(Mobile, 100_000_000, 8);
+decimal_scalar!(Generic, 100_000_000, 8);
```
Comment (madninja) [inline src/models/values.rs:120]: The `Generic` term is highly overloaded.. Perhaps just call it `Token`?

**Review by madninja — COMMENTED** [review]
looks good.. So this fails the build constantly?

**Comment by madninja** [comment]
> > > looks good.. So this fails the build constantly?
> > 
> > 
> > Ah I see it's because of the json api?
> 
> Yes, the API for master does not expose the new balances on accounts.

look at the rest of the errors? Looks like richest and get fail too

**Comment by lthiery** [comment]
> > > > looks good.. So this fails the build constantly?
> > > 
> > > 
> > > Ah I see it's because of the json api?
> > 
> > 
> > Yes, the API for master does not expose the new balances on accounts.
> 
> look at the rest of the errors? Looks like richest and get fail too

Those fails all have the same root: account model is updated here to include iot_balance and mobile_balance, but the API does not have that yet.

**Comment by lthiery** [comment]
> > > > looks good.. So this fails the build constantly?
> > > 
> > > 
> > > Ah I see it's because of the json api?
> > 
> > 
> > Yes, the API for master does not expose the new balances on accounts.
> 
> look at the rest of the errors? Looks like richest and get fail too

Those fails all have the same root: account model is updated here to include iot_balance and mobile_balance, but the API does not have that yet.

**Comment by lthiery** [comment]
I realized I could create default values for these fields.

**Review by madninja — COMMENTED** [review]
why not have Default for those types?

**Comment by lthiery** [comment]
> why not have Default for those types?

Ah true - that's better.

**Comment by lthiery** [comment]
I've completed testing used with:

- helium-wallet-rs: https://github.com/helium/helium-wallet-rs/pull/265
- helium-app-cli: https://github.com/helium/helium-ledger-cli/pull/21/files

**Comment by lthiery** [comment]
Old transactions in the ETL do not provide a `token_type`. When using `serde(tag)`, it doesn't know how to handle an untagged default unless you nest a tagged enum within an untagged enum. Rather than proliferating enums, a custom deserializer also fixes the problem.

**Review by madninja — APPROVED** [review]

**Pull request merged**